File: packages/rules/test/fixtures/twilio/py/library-verified.positive.py
import os

from flask import Flask, request
from twilio.request_validator import RequestValidator

app = Flask(__name__)
validator = RequestValidator(os.environ["TWILIO_AUTH_TOKEN"])


@app.route("/webhooks/twilio", methods=["POST"])
def twilio_webhook():
    url = request.url
    signature = request.headers.get("X-Twilio-Signature", "")
    if not validator.validate(url, request.form.to_dict(), signature):
        return "", 403
    return "", 200

File: packages/rules/test/fixtures/twilio/js/library-verified.positive.ts
import express from "express";
import twilio from "twilio";

const app = express();
app.use(express.urlencoded({ extended: false }));

app.post("/webhooks/twilio", async (req, res) => {
  const signature = (req.header("X-Twilio-Signature") ?? "") as string;
  const url = `https://example.com${req.originalUrl}`;
  const valid = twilio.validateRequest(
    process.env.TWILIO_AUTH_TOKEN ?? "",
    signature,
    url,
    req.body,
  );
  if (!valid) return res.status(403).end();
  res.status(200).send("<Response/>");
});

app.listen(3000);
